op-deployer/pkg/deployer/opcm/implementations2_test.go

```diff
@@ -0,0 +1,150 @@
+		require.NotNil(t, output)
+
+		// Now we run the old deployer
+		//
+		// We run it on a fresh host so that the deployer nonces are the same
+		// which in turn means we should get identical output
+		host2 := createTestHost(t)
+
+		// We'll need some contracts already deployed for this to work
+		proxyAdminAddress, proxyAddress, protocolVersionsAddress = deployDependencies(host2)
+
+		deprecatedOutput, err := opcm.DeployImplementations(host2, opcm.DeployImplementationsInput{
+			WithdrawalDelaySeconds:          big.NewInt(1),
+			MinProposalSizeBytes:            big.NewInt(2),
+			ChallengePeriodSeconds:          big.NewInt(3),
+			ProofMaturityDelaySeconds:       big.NewInt(4),
+			DisputeGameFinalityDelaySeconds: big.NewInt(5),
+			MipsVersion:                     big.NewInt(1),
+			// Release version to set OPCM implementations for, of the format `op-contracts/vX.Y.Z`.
+			L1ContractsRelease:    "dev-release",
+			SuperchainConfigProxy: proxyAddress,
+			ProtocolVersionsProxy: protocolVersionsAddress,
+			SuperchainProxyAdmin:  proxyAdminAddress,
+			UpgradeController:     common.BigToAddress(big.NewInt(13)),
+		})
+
+		// Make sure it succeeded
+		require.NoError(t, err)
+		require.NotNil(t, deprecatedOutput)
+
+		// Now make sure the addresses are the same
+		require.Equal(t, deprecatedOutput.AnchorStateRegistryImpl, output.AnchorStateRegistryImpl)
+		require.Equal(t, deprecatedOutput.DelayedWETHImpl, output.DelayedWETHImpl)
+		require.Equal(t, deprecatedOutput.DisputeGameFactoryImpl, output.DisputeGameFactoryImpl)
+		require.Equal(t, deprecatedOutput.ETHLockboxImpl, output.ETHLockboxImpl)
+		require.Equal(t, deprecatedOutput.L1CrossDomainMessengerImpl, output.L1CrossDomainMessengerImpl)
+		require.Equal(t, deprecatedOutput.L1ERC721BridgeImpl, output.L1ERC721BridgeImpl)
+		require.Equal(t, deprecatedOutput.L1StandardBridgeImpl, output.L1StandardBridgeImpl)
+		require.Equal(t, deprecatedOutput.MipsSingleton, output.MipsSingleton)
+		require.Equal(t, deprecatedOutput.Opcm, output.Opcm)
+		require.Equal(t, deprecatedOutput.OpcmContractsContainer, output.OpcmContractsContainer)
+		require.Equal(t, deprecatedOutput.OpcmDeployer, output.OpcmDeployer)
+		require.Equal(t, deprecatedOutput.OpcmGameTypeAdder, output.OpcmGameTypeAdder)
+		require.Equal(t, deprecatedOutput.OpcmInteropMigrator, output.OpcmInteropMigrator)
+		require.Equal(t, deprecatedOutput.OpcmUpgrader, output.OpcmUpgrader)
+		require.Equal(t, deprecatedOutput.OptimismMintableERC20FactoryImpl, output.OptimismMintableERC20FactoryImpl)
+		require.Equal(t, deprecatedOutput.OptimismPortalImpl, output.OptimismPortalImpl)
+		require.Equal(t, deprecatedOutput.PreimageOracleSingleton, output.PreimageOracleSingleton)
+		require.Equal(t, deprecatedOutput.ProtocolVersionsImpl, output.ProtocolVersionsImpl)
+		require.Equal(t, deprecatedOutput.SuperchainConfigImpl, output.SuperchainConfigImpl)
+		require.Equal(t, deprecatedOutput.SystemConfigImpl, output.SystemConfigImpl)
```

Comment:
Would a simple `reflect.DeepEqual(deprecatedOutput, output)` do the job here?
Have you considered whether it'd be worth implementing an equality checker to make this easier and safer?

Reply:
`require.JsonEq` might also work here

Reply:
I can try using `reflect.DeepEqual(deprecatedOutput, output)` - keeping in mind though that all of this code is going to be deleted soon (since we'll be dropping the old deployment scripts), I am not too keen on implementing a custom matcher

Reply:
Okay so `reflect.DeepEqual(deprecatedOutput, output)` does not work unfortunately since these are two different structs :(

Reply:
Also the general problem with `reflect.DeepEqual` is that you get a `true`/`false` which means if something fails, you need to go and `Printf` to find the difference, not ideal for CI failures

Reply:
I guess if you're deleting the code then it's no big deal.
Thanks for checking